### PR TITLE
Add torch-mlir examples

### PR DESCRIPTION
**Pull request by jorendumoulin**
This pull request adds some examples for converting torch models to MLIR

### CODE REVIEW

**Comment by jorendumoulin** [comment]
This PR is also awaiting the PR which includes a stardew installation in the docker container

**Review by JosseVanDelm — CHANGES_REQUESTED** [review]
Can you please add:
* [ ] instructions on how to run these examples in the README.md?
* [ ] a simple CI workflow that checks these examples with pytest? Simple examples are included in the stardew repo

File: tests/torch/test_matmul.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from stardew.torch import torch_compiler
+import pytest
+import torch
+
+def test_mult():
```
Comment (JosseVanDelm) [inline tests/torch/test_matmul.py:5]:
```suggestion
def test_matmul():
```

File: torch/matmul-relu-bias/main.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# but torch-mlir does not support non floating-point types for
+# the RELU activation function
+
+x = torch.randint(-10, 10, (256, 128), dtype=torch.float32)
+w = torch.randint(-10, 10, (128,  10), dtype=torch.float32)
+b = torch.randint(-10, 10, ( 10,    ), dtype=torch.float32)
+
+@torch_compiler(example=[x, w, b])
+def foo(x : torch.Tensor, w : torch.Tensor, b: torch.Tensor):
+    return torch.relu(x @ w + b)
```
Comment (JosseVanDelm) [inline torch/matmul-relu-bias/main.py:15]: Let's just run this as a test and an example at the same time?
That way we reduce code duplication :wink:

**Review by JosseVanDelm — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -48,3 +53,14 @@ Note: We use a `tollvm12.py` script for multiple reasons:
...
 More information is available [here](https://llvm.org/docs/OpaquePointers.html). 
+## Run torch-mlir tests
+
+The folder torch include some examples of translating torch models to mlir using the stardew framework.
+The python3.11 installation comes with all the requirements pre-installed and can be run:
+
+```sh
+cd /torch/mult
+python3 main.py
+```
```
Comment (JosseVanDelm) [inline README.md:64]: Please update after this run of the docker container

**Review by JosseVanDelm — APPROVED** [review]
See last comment

File: README.md (hunk 3 above)
Comment (jorendumoulin) [reply README.md]: see changes in new commit

**Pull request merged**